src/app/services/actions/samples-action-creators.ts

```diff
@@ -29,24 +30,20 @@ export function fetchSamples(): Function {
 
     const headers = {
       'Content-Type': 'application/json',
+      'Accept-Language': geLocale
     };
 
     const options: IRequestOptions = { headers };
 
     dispatch(fetchSamplesPending());
 
-    return fetch(samplesUrl, options)
-      .then(res => res.json())
-      .then(res => {
-        if (res.error) {
-          throw (res.error);
-        }
-        dispatch(fetchSamplesSuccess(res.sampleQueries));
-        return res.products;
-      })
-      .catch(error => {
-        dispatch(fetchSamplesError(error));
-      });
-
+    const response = await fetch(samplesUrl, options);
```

Comment:
When this `await` throws an error, `fetchSamplesError` won't be dispatched.

Reply:
I was facing an errror 500 from the API and noticed that as long as the response is not ok, it will displatch fetchSamplesError. Do you know if there would be other instances that we should be weary of?